tests/js/t14127_cast_number.nim

```diff
@@ -0,0 +1,6 @@
+doAssert cast[int32](-1) is int32
```

Comment:
add on top:
```nim
# bug #14127
```
for consistency with other tests / searchability (even if it's in filename t14127)

Reply:
I move to `tests/types/t14127_cast_number.nim` with multiple targets.